pcdsdevices/dg645.py

```diff
@@ -0,0 +1,447 @@
+            self.amplitude(4.0)
+        elif val=='OFF':
+            self.amplitude(0.0)
+
+    def trig_source(self, val=None):
+        if val is None:
+            return self.trigCh.trig_source
+        else:
+            self.trigCh.trig_source=val
+
+    def trig_inhib(self, val=None):
+        if val is None:
+            return self.trigCh.trig_inhibit
+        else:
+            self.trigCh.trig_inhibit=val
+
+
+class delayChannel(Device):
+    """
+    Class that defines an output (delay) channel of a DG645 delay generator.
+    
+    Parameters
+    ---------
+    prefix : ``str``
+        The PV base of the relevant DG, i.e 'MEC:LAS:DDG:01:'
+    
+    channel : ``str``
+        The delay channel on the DG, i.e. 'a', 'g', etc
+
+    name : ``str``                                                               
+        Alias for the delay generator
+    
+    delay_limits : ``tuple``, optional
+        Limits on the allowed delay in seconds. By default, the
+        limits are set to (0.0, 1.0).
+    """
+    delay_AO    = Cpt(EpicsSignal, 'DelayAO')
+    delay_SI    = Cpt(EpicsSignalRO,'DelaySI')
+    delay_Tweak = Cpt(EpicsSignal, 'DelayTweakAO')
+    delay_Inc   = Cpt(EpicsSignal, 'DelayTweakIncCO.PROC')
+    delay_Dec   = Cpt(EpicsSignal, 'DelayTweakDecCO.PROC')
+    ref_MI      = Cpt(EpicsSignal, 'ReferenceMI')
+    ref_MO      = Cpt(EpicsSignal, 'ReferenceMO')
+    
+    def __init__(self, prefix, channel, delay_limits=(0.0, 1.0), name='DelayChannel', **kwargs):
+        self.low_lim = delay_limits[0]
+        self.high_lim = delay_limits[1]
+        self.channel = channel
+        self.__refsDict = {"T0":0, "A":1, "B":2, "C":3, "D":4, "E":5, "F":6, 
+                           "G":7, "H":8}
+        self.__refsDictInv = {0:"T0", 1:"A", 2:"B", 3:"C", 4:"D", 5:"E", 6:"F", 
+                              7:"G", 8:"H"}
+        super().__init__(prefix=prefix+channel.lower(), name=name)
+        
+    def __call__(self, delay=None):
+        """
+        If DG645 instance is called with no attribute returns
+        or sets current delay of channel.
+        Usage
+        -----
+        dg645()      : reads back current delay
+        dg645(delay) : puts value in delay()
+        """
+        if delay is None:
+            return self.delay
+        else:
+            self.delay = delay
+            
+    @property
+    def delay(self):
+        """
+        All values are in seconds. The returned values are those
+        directly read off the DG645.
+        If value is None:
+        Returns the current delay value on the DG645 channel
+        If value is not None:
+        The delay value on the DG645 channel is changed to that
+        value.        
+        """
+        return float(self.delay_SI.get()[4:])
+        
+    @delay.setter
+    def delay(self, value):
+        checkValue(value, self.low_lim, self.high_lim)
+        self.delay_AO.put(value)
+        
+    def tweak(self, value=None, inc=False, dec=False):
```

Comment:
This is the same name as a motor function that enters a mode where the arrow keys are used to step up/down. This could potentially be confusing.